app/assets/javascripts/discourse/components/select-box.js.es6

```diff
@@ -0,0 +1,179 @@
+    if(this.get("expanded")) {
+      this.set("lastHoveredId", this.get("value"));
+
+      if(!this.get("_renderBody")) {
+        this.set("_renderBody", true);
+      };
+
+      Ember.run.scheduleOnce("afterRender", this, () => {
+        this.$(".select-box-filter .filter-query").focus();
+        this.$(".select-box-collection").css("max-height", this.get("maxCollectionHeight"));
+        this.$().removeClass("is-reversed");
+
+        const offsetTop = this.$()[0].getBoundingClientRect().top;
+        const windowHeight = Ember.$(window).height();
+        const headerHeight = this.$(".select-box-header").outerHeight();
+        const filterHeight = this.$(".select-box-filter").outerHeight();
+
+        if(windowHeight - (offsetTop + this.get("maxCollectionHeight") + filterHeight + headerHeight) < 0) {
+          this.$().addClass("is-reversed");
+          this.$(".select-box-body").css({
+            left: this.get("horizontalOffset"),
+            top: "",
+            bottom: headerHeight + this.get("verticalOffset")
+          });
+        } else {
+          this.$(".select-box-body").css({
+            left: this.get("horizontalOffset"),
+            top: headerHeight + this.get("verticalOffset"),
+            bottom: ""
+          });
+        }
+
+        this.$(".select-box-wrapper").css({
+          width: this.get("maxWidth"),
+          display: "block",
+          height: headerHeight + this.$(".select-box-body").outerHeight()
+        });
+      });
+    } else {
+      this.$(".select-box-wrapper").hide();
+    };
+  },
+
+  willDestroyElement() {
+    this._super();
+
+    Ember.$(document).off("click.select-box");
+  },
+
+  didReceiveAttrs() {
+    this._super();
+
+    this.set("lastHoveredId", this.get("data")[this.get("idKey")]);
+    this.set("filteredData", this._remapData(this.get("data")));
+    this._setSelectedData(this.get("data"));
+  },
+
+  didRender() {
+    this._super();
+
+    this.$(".select-box-body").css('width', this.get("maxWidth"));
+    this._expand();
+  },
+
+  didInsertElement() {
+    this._super();
+
+    Ember.$(document).on("click.select-box", (event) => {
+      if(this.get("expanded") && $(event.target).parents(".select-box").attr("id") !== this.$().attr("id")) {
+        this.set("expanded", false);
+      }
+    });
+  },
+
+  actions: {
+    onExpand() {
+      this.toggleProperty("expanded");
+    },
+
+    onFilterChange(filter) {
+      this.set("filter", filter);
+    },
+
+    onSelectRow(id) {
+      this.setProperties({
+        value: id,
+        expanded: false
+      });
+    },
+
+    onHoverRow(id) {
+      this.set("lastHoveredId", id);
+    }
+  },
+
+  _setSelectedData(data) {
+    const selectedData = _.find(data, (d)=> {
+      return d[this.get("idKey")] === this.get("value");
+    });
+
+    if(!_.isUndefined(selectedData)) {
+      this.set("selectedData", this._normalizeData(selectedData));
+    }
+  },
+
+  _remapData(data) {
+    return _.map(data, (d) => {
```

Comment:
You could do this in one  line since an arrow method returns by default. `return _.map(data, d => this._normalizeData(d))`

You could also just use regular javascript instead of underscore for the map, although I am less picky about that:

`data.map(d => this._normalizeData(d))`